### PR TITLE
KIL-2946: Generate API docs for the agent

### PR DESCRIPTION
**Pull request by mrostan**
- Added API docs using [flask-swagger](https://github.com/getsling/flask-swagger).
- New Makefile used by `circleci` build that:
  - Uses the CLI command from `flask-swagger` to generate the JSON API doc 
  - Uploads it to the S3 bucket used for hosting the static docs along the required `swagger-ui` files for showing the docs.
- This static website requires updating the CSP Security header (in CloudFront) to add `connect-src 'self'`
- Docs can be explored "live" (and endpoints tried out) by opening this page in the local dev environment: http://localhost:8081/swagger/. 
- Dev docs are available at: https://apollodocs.dev.getmontecarlo.com/

### CODE REVIEW

**Review by pxc-dev — APPROVED** [review]
Very nice! Left a few comments and suggestions.

I'll go ahead and create the infra resources for https://apollodocs.getmontecarlo.com and ping you on slack when deployed.

File: README.md [hunk 1]
```diff
@@ -98,6 +98,20 @@ gcloud run deploy dev-apollo-agent --image montecarlodata/pre-release-agent:late
...
+for production builds.
+In order to get better documentation, endpoints supporting multiple methods (like `GET` and `POST`) are 
+implemented using two methods to document the required parameters in the right way for each method. 
+The response element is defined in one of them and re-used in the other one, see `test_health` 
+in `generic/main.py` for an example.
+
+When running the agent locally, docs can be accessed using http://localhost:8081/swagger/ allowing endpoints to
+be tried out.
+For Azure, docs can also be accessed using the `/swagger` path, this works the same as the local development
+server but with security configured so `swagger-ui` allows you to set the App Key used for authentication.
```
[the remark this answers is not in the document]
Comment (mrostan) [reply README.md:113]: Good catch! Thanks

File: apollo/interfaces/azure/main.py [hunk 2]
```diff
@@ -19,6 +19,18 @@
...
+    "securityDefinitions": {
+        "Azure App Key": {
+            "type": "apiKey",
+            "name": "x-functions-key",
+            "in": "header",
+            "description": "Enter the Azure Function App Key.",
+        }
+    },
+    "security": [{"Azure App Key": []}],
+}
```
[the remark this answers is not in the document]
Comment (mrostan) [reply apollo/interfaces/azure/main.py:32]: Yup, good catch

File: requirements.in [hunk 3]
```diff
@@ -6,6 +6,7 @@ databricks-sql-connector==2.8.0
 dataclasses-json==0.6.0
 flask==2.3.3
 flask-compress==1.14
+flask-swagger==0.2.14
```
[the remark this answers is not in the document]
Comment (mrostan) [reply requirements.in]: Yup, moved to requirements-dev, thanks

File: apollo/interfaces/azure/main.py [hunk 4]
```diff
@@ -29,24 +41,146 @@ def azure_filter_extra(extra: Optional[Dict]) -> Optional[Dict]:
 main.logging_utils.extra_filterer = azure_filter_extra
 
 
-@app.route("/api/v1/azure/logs/query", methods=["GET", "POST"])
-def azure_logs_query() -> Tuple[Dict, int]:
+@app.route("/api/v1/azure/logs/query", methods=["GET"])
+def azure_logs_query_get() -> Tuple[Dict, int]:
+    """
+    Returns a list of Azure log events.
```
[the remark this answers is not in the document]
Comment (mrostan) [reply apollo/interfaces/azure/main.py:35]: [KIL-2983](https://linear.app/montecarlodata/issue/KIL-2983/add-linter-rules-to-enforce-docstring-for-rest-endpoints)

**Pull request merged**